### PR TITLE
docs: break installation instructions to collapsable sections 

### PR DESCRIPTION
**Pull request by matchai**
#### Description

As the number of supported systems and shells has grown, we've had to hand-pick what installation methods would be the "blessed" ones to show on our README. This change changes that to include _all_ installation methods, but hides them each into OS and shell-specific collapsed sections, as is done by the excellent [ajeetdsouza/zoxide](https://github.com/ajeetdsouza/zoxide).

**👉 [Preview the changes here](https://github.com/starship/starship/blob/rework-readme/README.md)**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -169,148 +169,230 @@
...
+]
+prompt = "starship_prompt"
+```
+
+</details>
+
+<details>
+<summary>PowerShell</summary>
+
+Add the following to the end of your PowerShell configuration (find it by running `echo $profile`):
```
Comment (rashil2000) [inline README.md:345]:
```suggestion
Add the following to the end of your PowerShell configuration (find it by running `$PROFILE`):
```

Minor nit: `echo` command in PowerShell is implicit. Variables can be directly queried. See more - https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/write-output?view=powershell-7.2#description

Comment (matchai) [reply README.md:345]: Ooh, thanks for the heads up, Rashil! 👍

File: README.md [hunk 2]
```diff
@@ -339,3 +421,19 @@ Please check out these previous works that helped inspire the creation of starsh
 
 Copyright © 2019-present, [Starship Contributors](https://github.com/starship/starship/graphs/contributors).<br>
 This project is [ISC](https://github.com/starship/starship/blob/master/LICENSE) licensed.
+
+[alpine linux packages]: https://pkgs.alpinelinux.org/packages?name=starship
+[arch linux community]: https://archlinux.org/packages/community/x86_64/starship
+[conda-forge]: https://anaconda.org/conda-forge/starship
```
Comment (rashil2000) [inline README.md:427]:
```suggestion
[chocolatey]: https://community.chocolatey.org/packages/starship
[conda-forge]: https://anaconda.org/conda-forge/starship
```

Starship is in community repository of Chocolatey

**Review by andytom — APPROVED** [review]
Love this it is so much easier to follow now, really clean. Do we want to remove the [advanced installation guide](https://starship.rs/installing/)?

**Comment by matchai** [comment]
Some of the more detailed, platform-specific details are still handy to have, like the Nix home-manager setup.
Most of the rest is probably redundant at this point though.

**Pull request merged**